### PR TITLE
Add Hashicorp Vault Kubernetes authentication method

### PR DESCRIPTION
**Pull request by liortamari**
Signed-off-by: liortamary <lior.tamary@houzz.com>

##### SUMMARY

Currently, Token and Approle authentication methods are supported for Hashicorp Vault.
When running on Kubernetes, it makes sense to support also the Kubernetes auth method.
https://www.vaultproject.io/docs/auth/kubernetes

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API

##### AWX VERSION

```
devel
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: awx/main/credential_plugins/hashivault.py [hunk 1]
```diff
@@ -47,15 +47,20 @@
             'multiline': False,
             'help_text': _('Name of the namespace to use when authenticate and retrieve secrets'),
         },
+        {
+            'id': 'kubernetes_role',
+            'label': _('Kubernetes role'),
+            'type': 'string',
+            'multiline': False,
+            'help_text': _('The Role for Kubernetes Authentication'),
```
Comment (shanemcd) [inline awx/main/credential_plugins/hashivault.py]: I'm wondering if we should include an example or point to some documentation here. It's not immediately clear to me what is expected for this field.

Comment (liortamari) [reply awx/main/credential_plugins/hashivault.py]: @shanemcd good idea, added.

**Comment by liortamari** [comment]
@shanemcd can you please assist to run the workflows after I fixed your review comments ?

**Comment by shanemcd** [comment]
@liortamari Apologies for the slow response. We recently added some new required CI checks. A quick rebase should get those pulled in.

**Comment by liortamari** [comment]
@shanemcd thanks for your help! 
I rebased, hope it will now be good for merge

**Comment by liortamari** [comment]
@shanemcd all checks passed now.

**Comment by shanemcd** [comment]
Awesome, nice work. @Zokormazo are you ok with merging this now or should we wait until after we cut the downstream release branch?

**Review by Zokormazo — APPROVED** [review]

**Pull request merged**